airflow/api_fastapi/execution_api/routes/__init__.py

```diff
@@ -17,10 +17,11 @@
 from __future__ import annotations
 
 from airflow.api_fastapi.common.router import AirflowRouter
-from airflow.api_fastapi.execution_api.routes import connections, health, task_instance, variables
+from airflow.api_fastapi.execution_api.routes import connections, health, task_instance, variables, xcoms
 
 execution_api_router = AirflowRouter()
 execution_api_router.include_router(connections.router, prefix="/connections", tags=["Connections"])
 execution_api_router.include_router(health.router, tags=["Health"])
 execution_api_router.include_router(task_instance.router, prefix="/task_instance", tags=["Task Instance"])
 execution_api_router.include_router(variables.router, prefix="/variables", tags=["Variables"])
+execution_api_router.include_router(xcoms.router, prefix="/xcoms", tags=["XComs"])
```

Comment:
We should be consistent on our pluralization - task_instance is singular, all the rest are plural

Reply:
Yeah I went back and forth on this. Reasoning was that `task_instance` was always ever going to operator on a single TI, while the other endpoints would operate on one or more of it (XCom, vars, conns etc).

I will make them singular for now.

Reply:
The single vs plural question in API design is one as old as REST.

Reply:
:) -- I changed my mind again -- going to make it plural — so I don’t have to change it in future if we add list, collection get endpoints and the fact that Public API has it plural too, so for code consistency!

Reply:
done in https://github.com/apache/airflow/pull/43894/commits/542a7034e18bbf35d1b7b6cc3cb538ad2ad0322b